runtime/storage/cache/cache.go

```diff
@@ -0,0 +1,267 @@
+// Package cache provides the ability to define distributed Redis cache clusters
+// and functionality to use them in a fully type-safe manner.
+//
+// For more information see https://encore.dev/docs/develop/caching
+package cache
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+)
+
+// ClusterConfig represents the configuration of cache clusters.
+type ClusterConfig struct {
+	// EvictionPolicy decides how the cache evicts existing keys
+	// to make room for new data when the max memory limit is reached.
+	//
+	// If not specified the cache defaults to AllKeysLRU.
+	EvictionPolicy EvictionPolicy
+}
+
+// An EvictionPolicy describes how the cache evicts keys to make room for new data
+// when the maximum memory limit is reached.
+//
+// See https://redis.io/docs/manual/eviction/#eviction-policies for more information.
+type EvictionPolicy string
+
+// NOTE: These values need to be added to the runtimeconstants package
+// and the parser package for the parser to be aware of them.
+
+const (
```

Comment:
I think again I'd prefer `LeastRecentlyUsed`, `LeastFrequentlyUsed` and `Random` here. it's easier to understand what the behaviour is without having to read the documentation of the contants.

Reply:
Yeah I hear you, but it's a bit of a mouthful: `AllKeysLeastFrequentlyUsed`.

I don't think the idea of splitting it into two fields is right as the underlying configuration is a single concept (the eviction policy) and not two orthogonal concepts.

It's not clear future policies will map cleanly to those two dimensions you described (and NoEviction arguably doesn't already today).

Reply:
> it's a bit of a mouthful: AllKeysLeastFrequentlyUsed.

sure, but with auto complete in IDE's it's not actually more keys - what it is though is more readable. With a good fuzzy autocomplete typing AllLFU will almost certainly auto complete the same thing